external/llvm-project/mlir/lib/Conversion/AMDGPUToROCDL/AMDGPUToROCDL.cpp

```diff
@@ -524,40 +524,40 @@ static std::optional<StringRef> mfmaOpToIntrinsic(MFMAOp mfma,
       return ROCDL::mfma_f64_4x4x4f64::getOperationName();
   }
 
-  if (sourceElem.isFloat8E5M2FNUZ() && destElem.isF32() &&
-      chipset.minorVersion >= 0x40) {
+  if ((sourceElem.isFloat8E5M2FNUZ() || sourceElem.isFloat8E5M2()) &&
```

Comment:
No

F8E45M2FNUZ is allowed on `minorVersion >= 0x40 && minorVersion < 0x50` and E5M2 and E4M3FN are on `minorVersion >= 0x50`